drivers/soundwire/intel.c

```diff
@@ -1051,6 +1051,24 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd, struct sn
 	return ret;
 }
 
+static int intel_component_probe(struct snd_soc_component *component)
+{
+	int ret;
+
+	/*
+	 * make sure the device is pm_runtime_active before initiating
+	 * bus transactions during the card registration
+	 */
+	ret = pm_runtime_resume_and_get(component->dev);
+	if (ret < 0 && ret != -EACCES)
```

Comment:
same comment here. pm_runtime_resume_and_get() will do a put_noidle() in case of -EACCES. Not sure that is intended at all

Reply:
and I also feel like almost all these changes are also wrong: https://github.com/thesofproject/linux/pull/3602/commits

Reply:
I also wonder if one of these changes is what is causing your original problem in the first place

Reply:
https://elixir.bootlin.com/linux/latest/source/drivers/base/power/runtime.c#L1121

The put_noidle decreases the value of usage_count that was incremented in resume, so that's fine I think.
it's the old code that was broken, on -EACCES the usage count was NOT decremented.